enhancements/installer/bootstrapping-clusters-with-capi-providers.md

```diff
@@ -0,0 +1,649 @@
+setup by [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest).
+
+### Workflow Description
+
+**cluster creator** is a human user responsible for deploying a
+cluster. Note that the workflow does not change for this user.
+
+**openshift-install** is the Installer binary.
+
+
+```mermaid
+sequenceDiagram
+  actor Operator
+  participant Installer
+  participant Local Control Plane
+
+  Operator->>Installer: provides `install-config` and credentials
+
+  opt customization
+    Operator->>+Installer: runs `openshift-install create manifests`
+    Installer-->>-Operator: generates manifests in the current folder
+    Operator-->>Operator: edits manifests
+  end
+
+  Operator->>+Installer: runs `openshift-install create cluster`
+  Installer->>Installer: extracts binaries for  to the install dir:<br/>kube-apiserver, etcd, <br/>CAPI infrastructure provider & cloud CAPI provider
+  Installer->>Local Control Plane: execs the CAPI infrastructure<br/> and cloud provider as subprocesses
+  Installer->>Local Control Plane: applies the CAPI manifests to the control plane
+
+  loop informer watch
+  Infra Controller-->Local Control Plane: Reconcile resources
+  end
+
+  Infra Controller->>+Cloud: provision cluster infrastrcuture
+  Cloud-->>Infra Controller: creation completed
+  Infra Controller->>Cloud: provision machines
+  Cloud-->>-Infra Controller: machines created
+
+  loop exponential backoff
+  Installer-->Local Control Plane: Wait for OpenShift Control Plane Availability
+  Infra Controller-->Cloud: checks bootstrap machine available
+  Installer-->OpenShift Cluster: waits until control plane is available and operators are healthy
+  end
+
+  Installer-->>-Operator: FIN
+```
+
+##### Error handling
+
+In the case of an error in the final step, the Installer will bubble up resources with non-expected statuses. The installer binary will collect and save all logs from all controllers used during the day zero operations, including Cluster API and its infrastructure providers.
+
+##### Credentials management
+
+The local control plane and Cluster API controllers share similar requirements regarding credentials when compared to Terraform. The installer validates and inject the required credentials from the user environments, cloud infrastructure providers can opt to pass credentials from the environment or install config in well-known Secret objects, or use the inherited environment available.
+
+#### Variation and form factor considerations [optional]
+
+
+### API Extensions
+
+As a result of this enhancement, API Extensions will only be used locally by the
+local Installer control plane--they will not, until future work, be pivoted to the cluster.
+The following are CRD references for Installer-supported platforms:
+
+- AWS: [infrastructure.cluster.x-k8s.io/v1beta2][awsCRD]
+- Azure/AzureStack: [infrastructure.cluster.x-k8s.io/v1beta1][azureCRD]
+- Baremetal: [infrastructure.cluster.x-k8s.io/v1beta1][metalCRD]
```

Comment:
Side note: CAPM3 is not currently a supported part of OpenShift and may have feature differences with CAPBM.

Reply:
I don't envisage us using CAPM3 at this stage, simply because the only thing it does on top of BMO is link the Machine to the Host, and Honza has already shown that we can do this statically without having to run any controller.